File: src/ProjectEuler/Problems/Problem2.hs
-- 4613732 - Completed 29.4.2013 - Learned list comprehensions.
module ProjectEuler.Problems.Problem2
    ( problem2 -- :: IO ()
    ) where

import ProjectEuler.Utils.Sequence (fibs)

problem2 :: IO ()
problem2 = print $ sum [x | x <- takeWhile (< 4000000) fibs, even x]

File: src/ProjectEuler/C/Problems/Problem2.c
// 4613732 - Completed 20.8.2013
#include <stdlib.h>
#include <stdio.h>

#include "../C_Utils/sequence.h"

int main(){
  int sum = 0,n,m;
    for (n = 3;1;n++){
	m = fib(n);
	if (m >= 4000000)
            break;
	else
	    sum = (m % 2 == 0) ? sum + m : sum;
    }
    printf("%d\n",sum);
    return 0;
}
